blueprints/addon/index.js

```diff
@@ -161,7 +161,23 @@ function alphabetizeDependencies(contents) {
 }
 
 function writeContentsToFile(contents, fileName) {
-  var packagePath = path.join(this.path, 'files', fileName);
+  var dirPath = path.join(this.path, 'files');
+  var packagePath = path.join(dirPath, fileName);
 
-  fs.writeFileSync(packagePath, stringifyAndNormalize(contents));
+  try {
+    fs.writeFileSync(packagePath, stringifyAndNormalize(contents));
```

Comment:
Most definitely. Do you think those files should be written directly into generated folder or stored in an os temp folder first and then moved after being generated?